src/bin/cli/radon.js

```diff
@@ -202,48 +224,63 @@ async function decode(flags, args, options) {
-                            templateArgs[index].push(`{:${templateArgs[index].length}}`)
-                        }
-                    })
-                    artifact = artifact.buildRequest(templateArgs)
-                    prefix = "RadonTemplate::"
 
-                } else if (artifact instanceof Witnet.RadonRetrieval) {
-                    if (artifact.argsCount > 0) {
-                        const retrievalArgs = artifactArgs.splice(0, artifact.argsCount)
-                        while (retrievalArgs.length < artifact.argsCount) {
-                            retrievalArgs.push(`{:${retrievalArgs.length}}`)
-                        }
-                        artifact = artifact.foldArgs(retrievalArgs)
-                    }
-                    artifact = new Witnet.RadonRequest({ retrieve: artifact })
-                    prefix = "RadonRetrieval::"
+        const assets = loadAssets(flags)
+        const headline = options?.headline
+        const crafts = flattenRadonArtifacts(assets).filter(craft => craft.key.toLowerCase().indexOf(asset.toLowerCase()) >= 0)
+        if (crafts.length === 0) {
+            throw `No matched found for pattern "${asset}"`
+        }
+        for (const index in crafts) {
+            const craft = crafts[index]
+            let artifact = craft.artifact
+            let prefix = ""
+            if (artifact instanceof Witnet.Radon.RadonRequest) {
+                prefix = "RadonRequest::"
+
+            } else if (artifact instanceof Witnet.Radon.RadonModal) {
+                artifact.providers = [ "https://dummy", ]
+                const modalArgs = []
+                let argIndex = 0
+                while (modalArgs.length < artifact.argsCount) {
+                    modalArgs.push(`{:${++ argIndex}}`)
                 }
-                if (!options?.headline) {
-                    options.headline = `${prefix}${craft.key}`
+                artifact = artifact.buildRadonRequest(modalArgs)
+                prefix = "RadonModal::"
+
+            } else if (artifact instanceof Witnet.Radon.RadonTemplate) {
+                const templateArgs = new Array(artifact.sources.length)
+                artifact.sources.forEach((retrieval, index) => {
+                    templateArgs[index] = []
+                    while (templateArgs[index].length < retrieval.argsCount) {
+                        templateArgs[index].push(`{:${templateArgs[index].length}}`)
+                    }
+                })
+                artifact = artifact.buildRadonRequest(templateArgs)
+                prefix = "RadonTemplate::"
+
+            } else if (artifact instanceof Witnet.Radon.RadonRetrieval) {
+                if (artifact.argsCount > 0) {
+                    const retrievalArgs = []
+                    let argIndex = 0
+                    while (retrievalArgs.length < artifact.argsCount) {
+                        retrievalArgs.push(`{:${++ argIndex}}`)
+                    }
```

Comment:
I'm not 100% this is equivalent for some edge case out of this scope but I think we are create an array here
```suggestion
                    const retrievalArgs = Array.from(
                      { length: artifact.argsCount },
                      (_, i) => `{:${i + 1}}`)
```